specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/managedCassandra.json

```diff
@@ -774,657 +693,1717 @@
+            "diskCapacity": {
               "type": "integer",
               "format": "int32",
-              "description": "Number of hours to wait between taking a backup of the cluster. To disable backups, set this property to 0."
+              "description": "Number of disk used for data centers. Default value is 4."
             },
-            "deallocated": {
+            "availabilityZone": {
               "type": "boolean",
-              "description": "Whether the cluster and associated data centers has been deallocated."
+              "description": "If the data center has Availability Zone support, apply it to the Virtual Machine ScaleSet that host the cassandra data center virtual machines."
             },
-            "cassandraAuditLoggingEnabled": {
+            "deallocated": {
               "type": "boolean",
-              "description": "Whether Cassandra audit logging is enabled"
+              "description": "Whether the data center has been deallocated."
+            },
+            "provisionError": {
+              "$ref": "#/definitions/CassandraError",
+              "description": "Error related to resource provisioning."
+            },
+            "authenticationMethodLdapProperties": {
+              "$ref": "#/definitions/AuthenticationMethodLdapProperties"
+            }
+          }
+        }
+      }
+    },
+    "CassandraClusterRepairPublicProperties": {
+      "type": "object",
+      "properties": {
+        "keyspace": {
+          "type": "string",
+          "description": "keyspace to be repaired."
+        },
+        "owner": {
+          "type": "string",
+          "description": "owner of the repair run."
+        },
+        "cause": {
+          "type": "string",
+          "description": "note on reason for repair run."
+        },
+        "tables": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "list of column famalies to be repaird. if empty whole keyspace will be repaired."
+        },
+        "segmentCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of segments in repair run"
+        },
+        "repairParallelism": {
+          "type": "string",
+          "description": "Defines the used repair parallelism for repair run. Valid values are SEQUENTIAL, PARALLEL or DATACENTER_AWARE."
+        },
+        "intensity": {
+          "type": "number",
+          "format": "double",
+          "description": "Defines the repair intensity for repair run."
+        },
+        "incrementalRepair": {
+          "type": "boolean",
+          "description": "Defines if incremental repair should be done. [true/false]. True when this flag is passed. False otherwise."
+        },
+        "autoStart": {
+          "type": "boolean",
+          "description": "Setting this flag will automatically start the repair run at time of creation."
+        },
```

Comment:
ARM recommends enums over booleans for future proof APIs.

Standard guidance is: replace boolean/switch properties with better enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum is more flexible and future proof option because it allows additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].. Also, properties should always ideally provide more meaningful values just than True and False, which are meaningless without context. For example two boolean switches "isTypeA" and "isTypeB" can be replaced with one enum "type": [A, B, DefaultType].

Note: do NOT just define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its still confusing, thanks to the boolean value names. Try to choose values that are meaningful even without context, like 'Encrypted', 'Unencrypted', 'TlsEncrypted'"